packages/desktop-client/src/components/reports/spreadsheets/net-worth-spreadsheet.ts

```diff
@@ -10,29 +11,35 @@ import {
   integerToAmount,
   amountToInteger,
 } from 'loot-core/src/shared/util';
+import {
+  type AccountEntity,
+  type RuleConditionEntity,
+} from 'loot-core/types/models';
 
-import { index } from '../util';
+type Balance = {
+  date: string;
+  amount: number;
+};
 
 export function createSpreadsheet(
-  start,
-  end,
-  accounts,
-  conditions = [],
-  conditionsOp,
+  start: string,
+  end: string,
+  accounts: AccountEntity[],
+  conditions: RuleConditionEntity[] = [],
+  conditionsOp: 'and' | 'or',
 ) {
-  return async (spreadsheet, setData) => {
-    if (accounts.length === 0) {
-      return null;
-    }
```

Comment:
This is the only functional change. Everything else is to make this file TS strict compliant.